src/types/events.d.ts

```diff
@@ -40,16 +48,47 @@ export interface Item {
   vendorId?: string;
 }
 
-export interface Purchase {
-  id: string;
+export interface Purchase extends BaseEvent {
   items: Item[];
-  occurredAt: string;
-  opaqueUserId: string;
+}
+
+interface BasePage {
+  pageId: string;
+}
+
+interface StandardPage extends BasePage {
+  type: "home" | "category" | "PDP" | "search" | "other";
+  value: string;
+}
+
+interface CartPage extends BasePage {
+  type: "cart";
+  value: string[];
+}
+
+/**
+ * Page information for a pageview event.
+ * Use StandardPage for single-value pages or CartPage for cart pages with multiple product IDs.
+ */
+export type Page = StandardPage | CartPage;
+
+/**
+ * PageView event - tracks when a user views a page.
+ */
+export interface PageView extends BaseEvent {
+  /** Channel through which the page was viewed */
+  channel?: ChannelType;
+  /** Type of device used */
```

Comment:
will this type of documenation work in editor hints? I think you should follow jsdoc instead